tools/nnyeah/nnyeah/Reworker.cs

```diff
@@ -570,20 +580,44 @@ Instruction ChangeMemberInstruction (Instruction instruction, IMemberDefinition
 			}
 		}
 
-		bool TryGetMethodTransform (Instruction instr, [NotNullWhen (returnValue: true)] out Transformation? result)
+		static bool IsCallInstruction (Instruction instr)
+		{
+			return instr.OpCode == OpCodes.Call ||
+				instr.OpCode == OpCodes.Calli ||
+				instr.OpCode == OpCodes.Callvirt;
+		}
+
+		bool TryGetMethodTransform (MethodBody body, Instruction instr, [NotNullWhen (returnValue: true)] out Transformation? result)
 		{
-			if (instr.OpCode != OpCodes.Call && instr.OpCode != OpCodes.Calli
-				&& instr.OpCode != OpCodes.Callvirt) {
+			if (!IsCallInstruction (instr)) {
 				result = null;
 				return false;
 			}
+
 			if (instr.Operand is MethodReference method && MethodSubs.TryGetValue (method.ToString (), out result)) {
 				return true;
 			}
 			result = null;
 			return false;
 		}
 
+		void PatchClassHandleReference (MethodBody body)
+		{
+			ILProcessor processor = body.GetILProcessor ();
+			for (int i = body.Instructions.Count - 1; i >= 0; i--) {
```

Comment:
Out of curiosity, why descending? (might be good to put the reason in case someone tries to be smart).

Reply:
I don't have full confidence that an insert won't break the code so that we end up doing infinite insertions since this changes the collection as work on it.